server/src/test/java/io/crate/planner/operators/JoinTest.java

```diff
@@ -1113,12 +1113,33 @@ public void test_push_constant_join_conditions_beneath_outer_joins() throws Exce
         QueriedSelectRelation mss = e.analyze("SELECT * FROM t1 RIGHT JOIN t2 ON t1.a = t2.b AND t1.a > 1");
 
         var plannerCtx = executor.getPlannerContext();
-        NestedLoopJoin result = (NestedLoopJoin) buildLogicalPlan(mss, plannerCtx);
+        var result = buildLogicalPlan(mss, plannerCtx);
+
+        assertThat(result).hasOperators(
+            "Eval[a, b]",
+            "  └ HashJoin[LEFT | (a = b)]",
```

Comment:
Maybe consider disabling hash joins for this test, since it's testing an irrelevant aspect, that (didn't verify) happens before the rewrite to hash join.

Reply:
👍